core-web/libs/sdk/client/src/lib/client/content/builders/collection/collection.ts

```diff
@@ -0,0 +1,334 @@
+import { Equals } from '../../../../query-builder/lucene-syntax';
+import { QueryBuilder } from '../../../../query-builder/sdk-query-builder';
+import { ClientOptions } from '../../../sdk-js-client';
+import { CONTENT_API_URL } from '../../shared/const';
+import {
+    GetCollectionResponse,
+    BuildQuery,
+    SortBy,
+    GetCollectionRawResponse,
+    GetCollectionError,
+    OnFullfilled,
+    OnRejected
+} from '../../shared/types';
+import { sanitizeQueryForContentType } from '../../shared/utils';
+
+/**
+ * Creates a Builder to filter and fetch content from the content API for an specific content type
+ *
+ * @export
+ * @class CollectionBuilder
+ * @template T Represents the type of the content type to fetch. Defaults to unknown
+ */
+export class CollectionBuilder<T = unknown> {
+    #page = 1;
+    #limit = 10;
+    #depth = 0;
+    #render = false;
+    #sortBy?: SortBy[];
+    #contentType: string;
+    #defaultQuery: Equals;
+    #query?: Equals;
+    #rawQuery?: string;
+
+    #serverUrl: string;
+    #requestOptions: ClientOptions;
+
+    constructor(requestOptions: ClientOptions, serverUrl: string, contentType: string) {
+        this.#requestOptions = requestOptions;
+        this.#serverUrl = serverUrl;
+
+        this.#contentType = contentType;
+
+        // We need to build the default query with the contentType field
+        this.#defaultQuery = new QueryBuilder().field('contentType').equals(this.#contentType);
+    }
+
+    /**
+     * This method returns the sort query in this format: field order, field order, ...
+     *
+     * @readonly
+     * @private
+     * @memberof CollectionBuilder
+     */
+    private get sort() {
+        return this.#sortBy?.map((sort) => `${sort.field} ${sort.order}`).join(',');
```

Comment:
What is the default here?

Reply:
`undefined` it means no sort at all

Reply:
The endpoint will always have sort... maybe we need to have one here... it could be something that all the the contrentlets will have like `modDate` or something like that.